### PR TITLE
PLAT-939 | Upgrade to php 7.3 and composer 1.8.4

### PR DESCRIPTION
**Pull request by rogatty**
https://bethink.atlassian.net/browse/PLAT-939

### CODE REVIEW

File: app/Console/Commands/TagsCleanup.php [hunk 1]
```diff
@@ -86,18 +86,18 @@ public function handle()
 					$this->info("Trimming tag {$name} to {$trimmed}");
 					$tag->name = $trimmed;
 					$tag->save();
-					continue;
+					break;
```
Comment (rogatty) [inline app/Console/Commands/TagsCleanup.php]: Fix for warning:
```
"continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"?
```
http://php.net/manual/en/control-structures.continue.php
>In PHP the switch statement is considered a looping structure for the purposes of continue. continue behaves like break (when no arguments are passed). If a switch is inside a loop, continue 2 will continue with the next iteration of the outer loop.

https://wiki.php.net/rfc/continue_on_switch_deprecation

**Comment by rogatty** [comment]
Smoke testing suggests everything works out of the box apart from the `continue` warning.

**Comment by kvas-damian** [comment]
Let's update dependency in `composer.json`

**Comment by kvas-damian** [comment]
```
/code/bethink/wnl-platform # ./vendor/bin/phpcs -p app --standard=PHPCompatibili
ty
............................................................  60 / 555 (11%)
............................................................ 120 / 555 (22%)
............................................................ 180 / 555 (32%)
............................................................ 240 / 555 (43%)
............................................................ 300 / 555 (54%)
............................................................ 360 / 555 (65%)
............................................................ 420 / 555 (76%)
............................................................ 480 / 555 (86%)
............................................................ 540 / 555 (97%)
...............                                              555 / 555 (100%)
```

**Comment by kvas-damian** [comment]
```
FILE: /code/bethink/wnl-platform/storage/framework/cache/facade-1b3d2202f839eba1159d1c004756ce641dfd30c9.php
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 1 | ERROR | An error occurred during processing; checking has been aborted. The error message was: Invalid testVersion setting: '7.3.2' in
   |       | /code/bethink/wnl-platform/vendor/phpcompatibility/php-compatibility/PHPCompatibility/Sniff.php on line 151
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------


FILE: /code/bethink/wnl-platform/vendor/phpunit/phpunit/tests/_files/SingletonClass.php
---------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
---------------------------------------------------------------------------------------
 20 | ERROR | Visibility for magic method __sleep must be public. Found: private
---------------------------------------------------------------------------------------


FILE: /code/bethink/wnl-platform/vendor/nikic/php-parser/test/PhpParser/ParserTest.php
------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
------------------------------------------------------------------------------------------------------------------------------------------------
 132 | ERROR | The behaviour of hexadecimal numeric strings was inconsistent prior to PHP 7 and support has been removed in PHP 7. Found: '0xf'
 133 | ERROR | The behaviour of hexadecimal numeric strings was inconsistent prior to PHP 7 and support has been removed in PHP 7. Found: '0XF'
------------------------------------------------------------------------------------------------------------------------------------------------


FILE: /code/bethink/wnl-platform/vendor/sentry/sentry/lib/Raven/Client.php
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 918 | ERROR | Since PHP 7.0, functions inspecting arguments, like debug_backtrace(), no longer report the original value as passed to a parameter, but will instead provide the current value. The
     |       | parameter "$data" was changed on line 849.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------


FILE: /code/bethink/wnl-platform/vendor/phpseclib/phpseclib/phpseclib/Crypt/Random.php
-------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 5 ERRORS AFFECTING 2 LINES
-------------------------------------------------------------------------------------------------------------------------------------------------------
  69 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
  69 | ERROR | Function mcrypt_create_iv() is deprecated since PHP 7.1 and removed since PHP 7.2; Use random_bytes() or OpenSSL instead
 108 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
 108 | ERROR | Function mcrypt_create_iv() is deprecated since PHP 7.1 and removed since PHP 7.2; Use random_bytes() or OpenSSL instead
 108 | ERROR | The constant "MCRYPT_DEV_URANDOM" is deprecated since PHP 7.1 and removed since PHP 7.2
-------------------------------------------------------------------------------------------------------------------------------------------------------


FILE: /code/bethink/wnl-platform/vendor/phpseclib/phpseclib/phpseclib/Crypt/Base.php
--------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 45 ERRORS AFFECTING 25 LINES
--------------------------------------------------------------------------------------------------------------------------------------------------------
  792 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
  792 | ERROR | Function mcrypt_generic_init() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
  825 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
  825 | ERROR | Function mcrypt_generic_init() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
  828 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
  828 | ERROR | Function mcrypt_generic() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
  833 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
  833 | ERROR | Function mcrypt_generic() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
  842 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
  842 | ERROR | Function mcrypt_generic() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
  852 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
  852 | ERROR | Function mcrypt_generic() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
  855 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
  855 | ERROR | Function mcrypt_generic_init() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
 1132 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
 1132 | ERROR | Function mcrypt_generic_init() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
 1160 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
 1160 | ERROR | Function mcrypt_generic() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
 1165 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
 1165 | ERROR | Function mcrypt_generic() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
 1174 | ERROR | Function mdecrypt_generic() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
 1177 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
 1177 | ERROR | Function mcrypt_generic_init() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
 1659 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
 1659 | ERROR | Function mcrypt_list_algorithms() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
 1738 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
 1738 | ERROR | Function mcrypt_module_close() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
 1739 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
 1739 | ERROR | Function mcrypt_module_close() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
 1744 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
 1744 | ERROR | Function mcrypt_module_close() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
 1851 | ERROR | The constant "MCRYPT_MODE_ECB" is deprecated since PHP 7.1 and removed since PHP 7.2
 1852 | ERROR | The constant "MCRYPT_MODE_CBC" is deprecated since PHP 7.1 and removed since PHP 7.2
 1854 | ERROR | The constant "MCRYPT_MODE_CFB" is deprecated since PHP 7.1 and removed since PHP 7.2
 1855 | ERROR | The constant "MCRYPT_MODE_NOFB" is deprecated since PHP 7.1 and removed since PHP 7.2
 1856 | ERROR | The constant "MCRYPT_MODE_STREAM" is deprecated since PHP 7.1 and removed since PHP 7.2
 1859 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
 1859 | ERROR | Function mcrypt_module_open() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
 1860 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
 1860 | ERROR | Function mcrypt_module_open() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
 1866 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
 1866 | ERROR | Function mcrypt_module_open() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
 1866 | ERROR | The constant "MCRYPT_MODE_ECB" is deprecated since PHP 7.1 and removed since PHP 7.2
 1871 | ERROR | Extension 'mcrypt' is deprecated since PHP 7.1 and removed since PHP 7.2; Use openssl (preferred) or pecl/mcrypt once available instead
 1871 | ERROR | Function mcrypt_generic_init() is deprecated since PHP 7.1 and removed since PHP 7.2; Use OpenSSL instead
--------------------------------------------------------------------------------------------------------------------------------------------------------


FILE: /code/bethink/wnl-platform/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/IBMDB2/DB2Statement.php
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 229 | ERROR | Since PHP 7.0, functions inspecting arguments, like func_get_args(), no longer report the original value as passed to a parameter, but will instead provide the current value. The parameter
     |       | "$fetchMode" was changed on line 218.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------


FILE: /code/bethink/wnl-platform/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/SQLAnywhere/SQLAnywhereStatement.php
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 214 | ERROR | Since PHP 7.0, functions inspecting arguments, like func_get_args(), no longer report the original value as passed to a parameter, but will instead provide the current value. The parameter
     |       | "$fetchMode" was changed on line 202.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------


FILE: /code/bethink/wnl-platform/vendor/symfony/http-kernel/Tests/DataCollector/MemoryDataCollectorTest.php
------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
------------------------------------------------------------------------------------------------------------------------------------------------
 48 | ERROR | The behaviour of hexadecimal numeric strings was inconsistent prior to PHP 7 and support has been removed in PHP 7. Found: '0x10'
 49 | ERROR | The behaviour of hexadecimal numeric strings was inconsistent prior to PHP 7 and support has been removed in PHP 7. Found: '0xf'
------------------------------------------------------------------------------------------------------------------------------------------------


FILE: /code/bethink/wnl-platform/vendor/symfony/var-dumper/Caster/ResourceCaster.php
-----------------------------------------------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 3 LINES
-----------------------------------------------------------------------------------------------------------
 66 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
 67 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
 68 | ERROR | Extension 'mysql_' is deprecated since PHP 5.5 and removed since PHP 7.0; Use mysqli instead
-----------------------------------------------------------------------------------------------------------


FILE: /code/bethink/wnl-platform/vendor/symfony/http-foundation/Tests/ParameterBagTest.php
--------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
--------------------------------------------------------------------------------------------------------------------------------------------------
 135 | ERROR | The behaviour of hexadecimal numeric strings was inconsistent prior to PHP 7 and support has been removed in PHP 7. Found: '0x100'
--------------------------------------------------------------------------------------------------------------------------------------------------


FILE: /code/bethink/wnl-platform/vendor/composer/composer/src/Composer/Util/ErrorHandler.php
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 63 | ERROR | Since PHP 7.0, functions inspecting arguments, like debug_backtrace(), no longer report the original value as passed to a parameter, but will instead provide the current value. The
    |       | parameter "$message" was changed on line 45.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------


FILE: /code/bethink/wnl-platform/vendor/dompdf/dompdf/src/Adapter/CPDF.php
----------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------------------
 198 | ERROR | Extension 'cpdf' is removed since PHP 5.1; Use pecl/pdflib instead
----------------------------------------------------------------------------------


FILE: /code/bethink/wnl-platform/vendor/hamcrest/hamcrest-php/tests/Hamcrest/Type/IsNumericTest.php
----------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------------------------------------------------------------------------------------
 28 | ERROR | The behaviour of hexadecimal numeric strings was inconsistent prior to PHP 7 and support has been removed in PHP 7. Found: '0x4F2a04'
----------------------------------------------------------------------------------------------------------------------------------------------------


FILE: /code/bethink/wnl-platform/vendor/phenx/php-svg-lib/src/Svg/Surface/SurfaceCpdf.php
-----------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
-----------------------------------------------------------------------------------------
 36 | ERROR | Extension 'cpdf' is removed since PHP 5.1; Use pecl/pdflib instead
-----------------------------------------------------------------------------------------


FILE: /code/bethink/wnl-platform/vendor/laravel/framework/src/Illuminate/Validation/Rule.php
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 47 | ERROR | Since PHP 7.0, functions inspecting arguments, like func_get_args(), no longer report the original value as passed to a parameter, but will instead provide the current value. The parameter
    |       | "$values" was changed on line 44.
 62 | ERROR | Since PHP 7.0, functions inspecting arguments, like func_get_args(), no longer report the original value as passed to a parameter, but will instead provide the current value. The parameter
    |       | "$values" was changed on line 59.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------


FILE: /code/bethink/wnl-platform/vendor/laravel/framework/src/Illuminate/Support/Collection.php
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  477 | ERROR | Since PHP 7.0, functions inspecting arguments, like func_get_args(), no longer report the original value as passed to a parameter, but will instead provide the current value. The
      |       | parameter "$keys" was changed on line 475.
 1255 | ERROR | Since PHP 7.0, functions inspecting arguments, like func_get_args(), no longer report the original value as passed to a parameter, but will instead provide the current value. The
      |       | parameter "$keys" was changed on line 1252.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------


FILE: /code/bethink/wnl-platform/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 845 | ERROR | Since PHP 7.0, functions inspecting arguments, like func_get_args(), no longer report the original value as passed to a parameter, but will instead provide the current value. The parameter
     |       | "$ids" was changed on line 842.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

Time: 7 mins, 8.13 secs; Memory: 158.01MB
```
All errors says that code should stop work on php7.2 already. Looks good!

**Pull request merged**